src/EditorFeatures/CSharpTest/ConvertProgram/ConvertToProgramMainTests.cs

```diff
@@ -0,0 +1,383 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using System;
+using System.Threading.Tasks;
+using Microsoft.CodeAnalysis.CodeStyle;
+using Microsoft.CodeAnalysis.CSharp;
+using Microsoft.CodeAnalysis.CSharp.CodeStyle;
+using Microsoft.CodeAnalysis.CSharp.ConvertProgram;
+using Microsoft.CodeAnalysis.CSharp.TopLevelStatements;
+using Microsoft.CodeAnalysis.Editor.UnitTests.CodeActions;
+using Xunit;
+
+namespace Microsoft.CodeAnalysis.Editor.CSharp.UnitTests.ConvertProgram
+{
+    using VerifyCS = CSharpCodeFixVerifier<ConvertToProgramMainDiagnosticAnalyzer, ConvertToProgramMainCodeFixProvider>;
+
+    public class ConvertToProgramMainTests
+    {
+        [Fact]
+        public async Task NotOfferedWhenUserPrefersTopLevelStatements()
+        {
+            var code = @"
+System.Console.WriteLine(0);
+";
+
+            await new VerifyCS.Test
+            {
+                TestCode = code,
+                FixedCode = code,
+                LanguageVersion = LanguageVersion.CSharp9,
+                TestState = { OutputKind = OutputKind.ConsoleApplication },
+                Options = { { CSharpCodeStyleOptions.PreferTopLevelStatements, true } },
+            }.RunAsync();
+        }
+
+        [Fact]
+        public async Task NotOfferedWhenUserPrefersProgramMainButNoTopLevelStatements()
+        {
+            var code = @"
+class C
+{
+    void M()
```

Comment:
Was this the intent of the test?

```suggestion
    void Main()
```

Reply:
nope.  the intent was as it was :)